src/acceptance-test/java/de/zalando/aruha/nakadi/webservice/PartitionsControllerAT.java

```diff
@@ -141,11 +142,16 @@ public void whenGetPartitionAndWriteMessageThenOffsetInPartitionIsIncreased() th
 
     private void validateOffsetIncreasedBy(final Map<String, String> partitionInfoBefore,
                                            final Map<String, String> partitionInfoAfter, final long delta) {
-        final long offsetBefore = Long.parseLong(partitionInfoBefore.get("newest_available_offset"));
-        final long offsetAfter = Long.parseLong(partitionInfoAfter.get("newest_available_offset"));
+        final long offsetBefore = getNewestOffsetAsLong(partitionInfoBefore);
+        final long offsetAfter = getNewestOffsetAsLong(partitionInfoAfter);
         assertThat(offsetAfter, is(offsetBefore + delta));
     }
 
+    private Long getNewestOffsetAsLong(final Map<String, String> partitionInfo) {
+        final String offset = partitionInfo.get("newest_available_offset");
+        return Cursor.BEFORE_OLDEST_OFFSET.equals(offset) ? -1 : Long.parseLong(offset);
```

Comment:
@rcillo I see your point, but to be consistent - then we should add the same comment to all other places in code where BEFORE_OLDEST_OFFSET is used.

@antban We agreed not to have too much acceptance tests, but try to cover more with unit tests. And in unit test all cases regarding offsets are covered (it's in https://github.com/zalando/nakadi/blob/nakadi-jvm/src/test/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepositoryTest.java)